### PR TITLE
Reduce visibility of prometheus libs

### PR DESCRIPTION
**Pull request by serathius**
Prevent references from kubelet and controller-manager

/kind feature

Ref https://github.com/kubernetes/enhancements/issues/1238

```release-note
NONE
```
/cc @RainbowMango

### CODE REVIEW

File: staging/src/k8s.io/component-base/metrics/BUILD [hunk 1]
```diff
@@ -88,14 +88,6 @@ package_group(
...
         "//cluster/images/etcd-version-monitor",
         "//pkg/controller/endpointslice",
-        "//pkg/controller/volume/attachdetach/metrics",
-        "//pkg/controller/volume/persistentvolume/metrics",
-        "//pkg/kubelet/metrics/collectors",
-        "//pkg/kubelet/pleg",
-        "//pkg/kubelet/pluginmanager/metrics",
-        "//pkg/kubelet/server",
-        "//pkg/kubelet/server/stats",
-        "//pkg/kubelet/volumemanager/metrics",
```
Comment (RainbowMango) [inline staging/src/k8s.io/component-base/metrics/BUILD]: Does this mean all the removed packages here have no reference to prometheus now?

**Comment by brancz** [comment]
Neat :+1: 

/lgtm
/approve

**Pull request merged**